tests/conftest.py

```diff
@@ -7908,3 +7915,65 @@ def finalizer():
 
     request.addfinalizer(finalizer)
     return factory
+
+
+@pytest.fixture(scope="session")
+def setup_cnv(request):
+    """
+    Session scoped fixture to setup and cleanup CNV
+    based on need of the tests
+
+    """
+    cnv_obj = CNVInstaller()
+    cnv_obj.deploy_cnv(check_cnv_deployed=True, check_cnv_ready=True)
+
+    def finalizer():
+        """
+        Clean up CNV deployment
+
+        """
+        if cnv_obj.cnv_hyperconverged_installed():
+            cnv_obj.cleanup_cnv()
+
+    request.addfinalizer(finalizer)
+
+
+@pytest.fixture()
+def setup_vms_standalone_pvc(request, project_factory, setup_cnv):
+    """
+    This fixture will setup VM using standalone PVC
+
+    """
+    vm_obj = None
+
+    def factory():
+        nonlocal vm_obj
+        project_obj = project_factory()
+        log.info(f"Created project {project_obj.namespace} for VMs")
+        vm_obj = create_vm_using_standalone_pvc(
+            running=True, namespace=project_obj.namespace
+        )
+        return vm_obj
+
+    def finalizer():
+        """
+        Teardown all the objects created as part of
+        the setup factory
+
+        """
+        pvc_obj = get_pvc_from_vm(vm_obj)
+        secret_obj = get_secret_from_vm(vm_obj)
+        volumeimportsource_obj = get_volumeimportsource(pvc_obj=pvc_obj)
+        vm_obj.delete()
+        log.info(f"Successfully deleted VM {vm_obj.name}")
+        pvc_obj.delete()
```

Comment:
we do not need to wait for the resource to delete?

Reply:
by default wait=True, so delete itself makes sure the resource is deleted. correct?

Reply:
No, the wait here is for the delete command to complete. It will not wait for the resource to delete. 
Please check https://github.com/mashetty330/ocs-ci/blob/cdc6601c397f41b37d0c2522de0a94035a2a8ae4/ocs_ci/ocs/ocp.py#L434

Reply:
```python
    def delete(self):
        """
        Delete the VirtualMachine
        """
        self.vm_ocp_obj.delete(resource_name=self._vm_name)
        self.vm_ocp_obj.wait_for_delete(resource_name=self._vm_name, timeout=180)
```